packages/monaco/src/browser/monaco-editor-provider.ts

```diff
@@ -238,7 +238,11 @@ export class MonacoEditorProvider {
     protected createOptions(prefixes: string[], uri: string, overrideIdentifier: string): { [name: string]: any };
     protected createOptions(prefixes: string[], uri: string, overrideIdentifier?: string): { [name: string]: any } {
         return Object.keys(this.editorPreferences).reduce((options, preferenceName) => {
-            const value = (<any>this.editorPreferences).get({ preferenceName, overrideIdentifier }, undefined, uri);
+            let value = (<any>this.editorPreferences).get({ preferenceName, overrideIdentifier }, undefined, uri);
+            // so monaco-editor can rewrite
```

Comment:
Is it because value is deep freeze? I wonder should not we do deep clone then? What kind of error do you get without it btw?

Reply:
Deep clone is better, I will modify this code.
Without this, we will get 'Cannot assign to read only property' error.

Reply:
fyi: https://github.com/theia-ide/theia/blob/9cf1895b7640029fbeb906b48172b093b7b0b91c/packages/core/src/common/objects.ts#L17